hydra-plutus/src/Hydra/Contract/MockHead.hs

```diff
@@ -53,23 +53,21 @@ hydraStateMachine _policyId =
   -- want as we need additional tokens being forged as well (see 'watchInit').
   SM.mkStateMachine Nothing hydraTransition isFinal
  where
-  -- XXX(SN): This is currently required to be able to observe the 'Abort'
-  -- transition!?
-  -- isFinal Final{} = True
+  isFinal Final{} = True
   isFinal _ = False
 
 {-# INLINEABLE hydraTransition #-}
 hydraTransition :: SM.State State -> Input -> Maybe (TxConstraints Void Void, SM.State State)
 hydraTransition oldState input =
   case (SM.stateData oldState, input) of
-    (Initial{}, CollectCom) ->
-      Just (mempty, oldState{SM.stateData = Open})
+    (Initial{}, CollectCom collectedValue) ->
+      Just (mempty, oldState{SM.stateData = Open, SM.stateValue = collectedValue <> SM.stateValue oldState})
```

Comment:
I think we have done this in a way which goes in the wrong direction. The real solution here would be to validate that the resulting value is `stateValue oldState` plus all the value of spent commit inputs.. which would be available to the validator in the `ScriptContext`?

Reply:
Correct. Will add a `FIXME` for this, but I guess we'll need to fix it anyway once we start writing real validators